v2/apiserver/internal/core/mongodb/events_store.go

```diff
@@ -370,13 +371,25 @@ func (e *eventsStore) CancelMany(
 	)
 	cur, err := e.collection.Find(ctx, criteria, findOptions)
 	if err != nil {
-		return events, errors.Wrapf(err, "error finding canceled events")
-	}
-	if err := cur.All(ctx, &events.Items); err != nil {
-		return events, errors.Wrap(err, "error decoding canceled events")
-	}
-
-	return events, nil
+		return nil, errors.Wrapf(err, "error finding canceled events")
+	}
+
+	eventCh := make(chan core.Event)
+	go func() {
+		defer close(eventCh)
+		for cur.Next(ctx) {
+			event := core.Event{}
+			if err := cur.Decode(&event); err != nil {
+				log.Println(errors.Wrap(err, "unable to decode event"))
+			}
+			select {
+			case eventCh <- event:
+			case <-ctx.Done():
```

Comment:
Another case where this context is tied to the inbound http request. If the client hangs up, cleanup will abort.